egs/wsj/s5/steps/nnet3/multilingual/allocate_multilingual_examples.py

```diff
@@ -0,0 +1,310 @@
+
+    egs.1.scp is generated using ranges.1.scp as following:
+    "num_examples" consecutive examples starting from line "local-scp-line" from
+    egs.scp file for language "source-lang" is copied to egs.1.scp.
+
+    To avoid loading whole scp files from all langs in memory,
+    egs.scp for is processed line by line using readline() for input langs
+    and to have more randomization across different archives,
+    num_jobs * num_archives temporary scp.job.archive_index files created in
+    egs/temp dir and all num_jobs scp.*.archive_index combined
+    into egs.archiv_index.scp.
+"""
+
+from __future__ import print_function
+import os, argparse, sys, random
+import logging
+import traceback
+
+sys.path.insert(0, 'steps')
+import libs.common as common_lib
+
+logger = logging.getLogger('libs')
+logger.setLevel(logging.INFO)
+handler = logging.StreamHandler()
+handler.setLevel(logging.INFO)
+formatter = logging.Formatter("%(asctime)s [%(pathname)s:%(lineno)s - "
+                              "%(funcName)s - %(levelname)s ] %(message)s")
+handler.setFormatter(formatter)
+logger.addHandler(handler)
+logger.info('Starts generating multilinual examples (allocate_multilingual_examples.py)')
+
+def get_args():
+
+    parser = argparse.ArgumentParser(description="Writes ranges.*, outputs.* and weights.* files "
+                                   "in preparation for dumping egs for multilingual training.",
+                                   epilog="Called by steps/nnet3/multilingual/get_egs.sh")
+    parser.add_argument("--samples-per-iter", type=int, default=40000,
+                        help="The target number of egs in each archive of egs, "
+                        "(prior to merging egs). ")
+    parser.add_argument("--num-jobs", type=int, default=20,
+                        help="This can be used for better randomness in distributing "
+                        "languages across archives, where egs.job.archive.scp generated "
+                        "randomly and examples are combined "
+                        "across all jobs as eg.archive.scp.")
+    parser.add_argument("--random-lang", type=str, action=common_lib.StrToBoolAction,
+                        help="If true, the lang-id in ranges.* selected"
+                        " w.r.t frequency distribution of remaining examples in each language,"
+                        " otherwise it is selected sequentially.",
+                      default=True, choices = ["false", "true"])
+    parser.add_argument("--max-archives", type=int, default=1000,
+                        help="max number of archives used to generate egs.*.scp")
+    parser.add_argument("--seed", type=int, default=1,
+                        help="Seed for random number generator")
+
+    parser.add_argument("--minibatch-size", type=int, default=512,
+                        help="The minibatch size used to generate scp files per job. "
+                             "It should be multiple of actual minibatch size.")
```

Comment:
It doesn't need to be a multiple of the actual minibatch size any more (since 5.1, nnet3-merge-egs will only merge egs with the same structure).  so you can remove this sentence.